### ISSUE TITLE
[Feature] Option for explicitly enabling Qalc in non .qalc documents

### ISSUE THREAD

**Issue by janogarcia** [nortakales/vs-code-qalc#35]
Thanks, Nick, for such a cool, useful extension!

For non .qalc documents, like Markdown, it would be awesome if we had a setting for enabling Qalc for specific blocks instead of parsing the entire document by default.

This increased granularity would prevent Qalc from unnecessarily parsing entire, large Markdown documents and, simultaneously, prevent it from doing unwanted calculations.

Making this available as a setting would be super welcome.

Example using HTML comments:

```markdown
# Aggregations
1
2
3
4
sum
avg

# Dates
<!-- qalc:on -->
3/14/2021 11:21:56 PM UTC
date(3/14/2021)
date(1615789316)
date(1615789316000)
yesterday
tomorrow
today
now
1 year in months
now - 3 days
3 days ago
<!-- qalc:off -->

# Variables and Objects
variable = 7
object = {field: 2}
object.field * variable
```

Alternatively, using code blocks could be even cleaner:

````markdown
# Aggregations
1
2
sum
avg

# Dates
```qalc
3/14/2021 11:21:56 PM UTC
date(3/14/2021)
date(1615789316)
```

# Variables and Objects
variable = 7
````

Live test using `//qalc:off` and `//qalc:on` by setting `let qalcEnabled = false;` by default in `MathDocument::evaluate()`:

https://github.com/nortakales/vs-code-qalc/blob/92ad4e6b9df17d4d8911e7b621fa7143726cdc7f/src/document.ts#L64

<img width="1136" alt="SCR-20240227-pldl" src="https://github.com/nortakales/vs-code-qalc/assets/60790/ccba8778-ba5b-48cf-81b2-cd39026e2643">

**Comment by nortakales** [comment]
For the sake of this discussion let's define a couple terms:

1. Enabled/disabled - whether or not Qalc looks at or does *anything* at all with a file
2. Processing on/off - whether or not Qalc is processing a specific line and outputting results

I am trying to think of how best to implement this from a user perspective. I can see at least two options:

1. Qalc will be enabled by default for all files, but processing will be on by default for only file types listed in `qalc.enabledLanguages`, and otherwise will be off by default with the ability to turn it on for sections via `//qalc:on`
2. `qalc.enabledLanguages` will exist as is and control the file types Qalc is enabled for, and there can be a new setting(s) to control which files or file types have processing on/off by default.

I think #2 gives a little more flexibility and maybe some performance gains as you can tell Qalc which files types to look at and to completely ignore certain file types (rather than searching for //qalc:on in *all* files)

**Comment by janogarcia** [comment]
If you don’t wantwan’t to overcomplicate the user settings and keep `qalc.enabledLanguages` as is, you could even consider starting out with just a checkbox. For example:

- [ ] **Explicit Qalc Invocation** <br> When turned on, Qalc will only process the content enclosed within `//qalc:on` and `//qalc:off` markers. This setting applies to all enabled languages other than `“qalc”`, which contents still get processed by default.

It should be easier to maintain from a user perspective. Set enabled languages, mark checkbox, and done without needing to keep track of a second array of languages.

In the event you want to turn that checkbox into an array in the future for even added granularity, it should be straightforward to automate that migration.

Just an idea.

**Comment by nortakales** [comment]
Addressed in version 0.4

**Comment by janogarcia** [comment]
Awesome, thanks for adding support for it!
